[Tan Wei Xin] iP

weixin-tan opened the pull request.

Commit:
Add Gradle support

weixin-tan committed:
Implement skeletal

weixin-tan committed:
Customise hello and goodbye

weixin-tan committed:
Add "Add" and "List" functionaility

weixin-tan committed:
Add Task class

weixin-tan committed:
Add "mark done" functionality

weixin-tan committed:
Add Deadline, Event, and Todos Task

weixin-tan committed:
Add ability to add different tasks

weixin-tan committed:
Fix illegal symbols

weixin-tan committed:
Update runtest.bat to compile all relevant .java files

weixin-tan committed:
Add method to generate test cases

weixin-tan committed:
Add error handling

weixin-tan committed:
Add ability to "delete" task and update test case generator accordingly

weixin-tan committed:
Add abillity to save

weixin-tan committed:
Fix save string for deadline and event tasks

weixin-tan committed:
Increase frequency of writes

weixin-tan committed:
Update gitignore

weixin-tan committed:
Merge branch 'branch-Level-7' into master

weixin-tan committed:
Add empty file to be used as database

weixin-tan committed:
Merge branch 'branch-Level-7' into master

weixin-tan committed:
Merge branch 'master' into branch-Level-8

weixin-tan committed:
Stop track database

weixin-tan committed:
Use LocalDate to store date formatted as yyyy-mm-dd

weixin-tan committed:
Remove redundancy

weixin-tan committed:
Stop tracking database

weixin-tan committed:
Merge branch 'branch-Level-8' into master

weixin-tan committed:
Make OOP

weixin-tan committed:
Package classes

weixin-tan committed:
Add ability to find tasks and list tasks with custom message

weixin-tan committed:
Merge branch 'branch-Level-9' into master

weixin-tan committed:
Add javadoc comments.

weixin-tan committed:
Improve code quality

weixin-tan committed:
Merge branch 'branch-A-javadoc' into master

weixin-tan committed:
Merge branch 'branch-A-CodingStandard' into master

weixin-tan committed:
Add Parser list and bye test

weixin-tan committed:
Prepare for JAR

KendrewChan commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,81 @@
+     *
+     * @param strings The collection of strings to be printed.
+     */
+    public void printNicelyCollection(Collection<String> strings) {
+        System.out.println("________________________________________");
+        for (String s: strings) {
+            System.out.println("    " + s);
+        }
+        System.out.println("________________________________________");
+    }
```
Good use of abstraction method `printNicelyCollection()`for the `"______"` lines as they may not be used and changed to something else in the future, which makes the code more adaptable. Could possibly use `String.format()` for a slightly cleaner code where you only use `System.out.println(String.format(""))` once

weixin-tan replied:
Partially fixed by abstracting out the border. Not too sure about the String.format() part. Can elaborate?

KendrewChan commented on src/main/java/duke/ToDoTask.java:
```diff
@@ -0,0 +1,27 @@
+    public String toString() {
+        return "[T]" + super.toString();
+    }
+
+
```
Unintended additional space

weixin-tan replied:
Fixed.

KendrewChan commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,53 @@
+/**
+ * Represents a list of tasks.
+ */
+public class TaskList extends ArrayList<Task> {
```
TaskList inherits from ArrayList which makes the entire codebase really clean

MorningLit replied:
+1! Beautiful code!!

KendrewChan commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,26 @@
+        this.isDone = true;
+    }
+
+    @Override
+    public String toString() {
+        return String.format("[%c] %s", isDone ? '\u2713' : '\u2717', desc);
+    }
```
Might be cleaner to abstract this out as `String isDoneStr = isDone ? '\u2713' : '\u2717'`

weixin-tan replied:
Fixed.

KendrewChan commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,65 @@
+package duke;
+
+import java.io.*;
```
CheckStyle convention recommends to import one by one

weixin-tan replied:
Fixed.

KendrewChan commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,154 @@
+    }
+
+    private static boolean isDone(String s) {
+        return s.length() >= 4 && s.substring(0, 4).toLowerCase().equals("done");
```
Would be good to place line break to remove readability

MorningLit replied:
remove readability, do you mean improve readability, kendrew chan?

weixin-tan replied:
Fixed similar parts.

KendrewChan commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,154 @@
+     * @return A PARSER_RESULT value.
+     * @throws IllegalArgumentException Unrecognizable command.
+     */
+    public PARSER_RESULT parseInput(String input) throws IllegalArgumentException {
+        input = input.trim();
+        if (isBye(input)) {
+            return PARSER_RESULT.BYE;
+        } else if (isDelete(input)) {
+            return PARSER_RESULT.DELETE;
+        } else if (isDone(input)) {
+            return PARSER_RESULT.DONE;
+        } else if (isList(input)) {
+            return PARSER_RESULT.LIST;
+        } else if (isFind(input)) {
+            return PARSER_RESULT.FIND;
+        } else if (isTask(input)) {
+            return PARSER_RESULT.ADD;
+        } else {
+            throw new IllegalArgumentException("Unrecognizable command.");
+        }
+    }
```
Good use of enums and methods to abstract out the logic to parse the `String input`

MorningLit replied:
+1, overall very clean and well-designed code

KendrewChan commented on src/main/java/duke/EventTask.java:
```diff
@@ -0,0 +1,48 @@
+    @Override
+    public String toSaveString() {
+        return String.format("E @@ %d @@ %s @@ %s",
+                isDone ? 1 : 0, desc, period == null ? periodString : period.toString());
```
Might be cleaner to abstract out the logic into `String variables` to improve readability

weixin-tan replied:
Fixed all similar parts.

KendrewChan reviewed (commented):
Looks really good, I learned a lot.
Some cosmetic changes required

MorningLit commented on src/main/java/duke/DeadlineTask.java:
```diff
@@ -0,0 +1,51 @@
+ * Represents a deadline task with a deadline.
+ */
+public class DeadlineTask extends Task {
+    LocalDate deadline;
```
Maybe set private and final for variables? Since you're not modifying it anymore

weixin-tan replied:
Resolved.

MorningLit commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,77 @@
+        }
+        ui.greet();
+        ui.listOut(taskList);
+        while (true) {
```
Try to avoid while (true) ? I think its generally bad practice? Can replace with boolean flags

weixin-tan replied:
Resolved.

MorningLit commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,77 @@
+            try {
+                String input = ui.nextLine();
+                switch (parser.parseInput(input)) {
+                case ADD:
```
Amazing use of enums

MorningLit reviewed (approved):
Generally, a very clean and efficient code! Also, demonstrates very good use of enums! There are only very few minor mistakes in the code which only requires a few minutes to correct. LGTM and very well done!

weixin-tan committed:
Fixed styling problems.

weixin-tan committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

weixin-tan committed:
Add Gradle and CheckStyle support

weixin-tan committed:
Merge branch 'branch-A-Gradle' into master

weixin-tan committed:
Update CheckStyle version

weixin-tan committed:
Display UTF-8 characters in IntelliJ

weixin-tan committed:
Add and update relevant classes for GUI to work.

weixin-tan committed:
Merge branch 'branch-GUI' into master

weixin-tan committed:
Change duke and user image

weixin-tan committed:
Add and enable assertions

weixin-tan committed:
Improve coding quality.

weixin-tan committed:
Merge pull request #2 from duckmoon99/branch-A-CodeQuality

Improve coding quality

weixin-tan committed:
Merge branch 'master' into branch-A-Assertions

weixin-tan committed:
Merge pull request #1 from duckmoon99/branch-A-Assertions

Add and enable assertions

weixin-tan committed:
Modify checkstyle to be more compatible with Lambdas and Streams

Disables check that "(" is at the previous end of line in a line wrap.

weixin-tan committed:
Fix wrong assertion

weixin-tan committed:
Modify syntax to be friendlier

Users can now type part of the command instead of the full one (eg. "l" instead of "list") as long as it is not ambiguous.

weixin-tan committed:
Update ParserTest to be stronger

weixin-tan committed:
Merge branch 'branch-C-FriendlierSyntax' into master

weixin-tan committed:
Add gradle.yml

weixin-tan committed:
Update gradle.yml

weixin-tan committed:
Merge pull request #3 from duckmoon99/branch-A-CI

Add CI

weixin-tan committed:
Remove redundant setup

weixin-tan committed:
Add product website

weixin-tan committed:
Set theme jekyll-theme-minimal
